engine/graphics/src/opengl/graphics_opengl.cpp

```diff
@@ -484,52 +484,51 @@ static void LogFrameBufferError(GLenum status)
 
     OpenGLContext* g_Context = 0x0;
 
-    static void UpdateGLHandle(HContext context, HOpenglID idx, GLuint handle)
+    static HOpenglID AddNewGLHandle(HContext _context, GLuint handle)
     {
-        OpenGLContext* gl_context = (OpenGLContext*)context;
-        gl_context->m_AllGLHandles[idx] = handle;
-    }
+        OpenGLContext* context = (OpenGLContext*) _context;
 
-    static HOpenglID AddNewGLHandle(HContext context, GLuint handle)
-    {
-        OpenGLContext* gl_context = (OpenGLContext*)context;
-        HOpenglID result_idx = gl_context->m_AllGLHandles.Size();
-        if (!gl_context->m_FreeIndexes.Empty())
+        ScopedLock lock(context->m_GLHandlesData.m_Mutex);
+        HOpenglID result_idx = context->m_GLHandlesData.m_AllGLHandles.Size();
+        if (!context->m_GLHandlesData.m_FreeIndexes.Empty())
         {
-            result_idx = gl_context->m_FreeIndexes.Back();
-            gl_context->m_FreeIndexes.Pop();
+            result_idx = context->m_GLHandlesData.m_FreeIndexes.Back();
+            context->m_GLHandlesData.m_FreeIndexes.Pop();
         }
         else
         {
-            if (gl_context->m_AllGLHandles.Full())
+            if (context->m_GLHandlesData.m_AllGLHandles.Full())
             {
-                gl_context->m_AllGLHandles.OffsetCapacity(32);
+                context->m_GLHandlesData.m_AllGLHandles.OffsetCapacity(32);
             }
-            gl_context->m_AllGLHandles.Push(0);
+            context->m_GLHandlesData.m_AllGLHandles.Push(0);
         }
 
-        UpdateGLHandle(context, result_idx, handle);
+        context->m_GLHandlesData.m_AllGLHandles[result_idx] = handle;
         return result_idx;
     }
 
     static inline GLuint GetGLHandle(OpenGLContext* context, HOpenglID idx)
     {
-        return context->m_AllGLHandles[idx];
+        ScopedLock lock(context->m_GLHandlesData.m_Mutex);
+        return context->m_GLHandlesData.m_AllGLHandles[idx];
     }
 
     static inline GLuint* GetGLHandlePointer(OpenGLContext* context, HOpenglID idx)
     {
-        return &(context->m_AllGLHandles[idx]);
+        ScopedLock lock(context->m_GLHandlesData.m_Mutex);
+        return &(context->m_GLHandlesData.m_AllGLHandles[idx]);
```

Comment:
This looks like an old bug.
Giving out a pointer to an array that may reallocate, will (silently) give you a dangling pointer.
Should be possible to catch using a test (making sure the array reallocates), and running with `--with-asan`